### ISSUE TITLE
Assembler, bin parser: Extended instructions allowed to have too many parameters

### ISSUE THREAD

**Issue by dneto0** [KhronosGroup/SPIRV-Tools#233]
The parser is allowing an extended instruction to have too many parameters.
For example, the OpenCL pow instruction is only supposed to have two normal arguments, but the following is accepted by the assembler, and the result is happily disassembled by the binary-parser/disassembler.
```
          %1 = OpExtInstImport "OpenCL.std"
          %blam  = OpExtInst %a %1 pow %x %y %z
          OpNop
```
The %z argument should have triggered an error.

**Comment by dneto0** [comment]
Also affects GLSL.std.450:
```
          %1 = OpExtInstImport "GLSL.std.450"
          %blam  = OpExtInst %a %1 Round %2 %3 %5 %7 %8
          OpNop
```
The Round extended instruction in GLSL is only supposed to take one normal argument.

**Comment by dneto0** [comment]
I traced it to the SPIR-V core instruction grammar file:  For OpExtInst, there is an argument following the LiteralExtInstInteger operand whic his IdRef with quantifier "*".  In the assembler and binary parser, the value of the extended instruction integer determines the expected operands, but then the IdRef-quantifier-* is still on the expected-operands stack.

In other cases, e.g. OpImageFetch, the ImageOperands bit mask fully determines the operands to follow and the grammar does *not* include them in the ImageFetch instruction listing.   It's arguable whether the same rule should apply in the case of OpExtInst.   Since we've already published the grammar, and it's arguable how much checking other tools should have to do, I'm ok just patching this in the SPIRV-Tools setup phase.
